### ISSUE TITLE
Treemacs don't show mode-line until re-enable doom-modeline-mode

### ISSUE THREAD

**Issue by EndlessPeak** [Alexander-Miller/treemacs#843]
As the title describe,I have tried to set the following code in my `~/.doom.d/config.el`

```elisp
(use-package! treemacs
  :ensure t
  :config
  (progn
    (setq treemacs-user-mode-line-format t)))
```

When I toggled treemacs,the mode-line didn't show.I tried to disable `doom-modeline-mode` and enable it again,it shows a mode-line for treemacs.But it seems like doom automatically set a mode-line for a normal window rather than treemacs customize it,because the mode-line not only show "Treemacs" caption.
In addition,it generated a error,which read as:

```elisp
Error during redisplay (eval (doom-modeline-segment--modals)) signnaled (void-function nil) 
```
Here is two screenshot,the former shows it cannot show the mode-line.

![2021-09-15_09-26](https://user-images.githubusercontent.com/53530553/133355611-039e0519-3125-4cb1-a031-2450cb283693.png)

The latter is generate after re-enable `doom-modeline-mode`.
![2021-09-15_09-27](https://user-images.githubusercontent.com/53530553/133355625-8a38cd24-e3d7-4bba-8205-6862e2635ab5.png)

That's all the information I've got,hope my poor english and my issue won't disturb you.

**Comment by bestlem** [comment]
This is doom-modeline's problem.

When it starts it sets the modeline for all buffers and overrides the existing treemacs one.

I think if you enable doom-modeline mode first and then load treemacs the treemacs modeline might win.

**Comment by Alexander-Miller** [comment]
What modeline were you trying to show exactly? Setting `treemacs-user-mode-line-format` to `t` will not do anything. If set its value is passed directly to `mode-line-format`. If you want to show a doom-modeline in treemacs that is not the basic default you 2 options:

- Use `treemacs-mode-hook` to call `(doom-modeline-set-modeline 'FOO)`
- Set `treemacs-user-mode-line-format` to `("%e" (:eval (doom-modeline-format--FOO)))`

where FOO is whatever doom modeline you want to see.

**Comment by EndlessPeak** [comment]
Not the doom one,I want to see treemacs' mode-line,I think its concise style for treemacs is more suitable for sidebar.
Last night I tried to follow @bestlem 's instruction,but don't know how to operation.Then I mistakenly commented treemacs and doom removed the attached packages of treemacs (such as treemacs-magit,I counted four,but not itself).
Then treemacs' modeline magically showed for itself.Quite weird,I think.
I tried use straight.el install those package,then the question happened again.
Since I don't need treemacs-magit and other attached packages now,in some sense,my question got solved,though I don't know why it works in such way.
Anyway,many thanks for your reply.
